hera_cal/vis_clean.py

```diff
@@ -213,6 +442,46 @@ def apply_calibration(self, input_cal, unapply=False):
         apply_cal.calibrate_in_place(self.data, cal_gains, self.flags, cal_flags,
                                      gain_convention=gain_convention)
 
+    def apply_flags(self, external_flags, overwrite_flags=False, type='uvflag'):
+        """
+        Apply external set of flags to self.hd.flag_array (inplace!), and re-attach the data and flags.
+        Default is to OR existing flags, unless overwrite_flags is True.
+
+        Parameters
+        ----------
+        external_flags: str or UVFlag object
+            Str or list of strings pointing to flag files to apply.
+            flag files should be in a format readable by UVFlag.
+        overwrite_flags: bool, optional
+            If True, overwrite flags (instead of OR) for baselines that are not entirely flagged.
+        type : str, optional
+            Use 'yaml' if the flags are a yaml file or 'uvflag' if the flags are a UVFlag
+            file or object. Default is 'uvflag'.
+        """
+
+        # get a census of fully flagged baselines to re-flag if overwrite_flags
+        if overwrite_flags:
+            full_flags = [bl for bl in self.flags if np.all(self.flags[bl])]
+
+        # apply flags
+        if type == 'uvflag':
+            if isinstance(external_flags, str):
+                external_flags = UVFlag(external_flags)
+            uvutils.apply_uvflag(self.hd, external_flags, unflag_first=overwrite_flags, inplace=True)
+        elif type == 'yaml':
+            from hera_qm.utils import apply_yaml_flags
+            self.hd = apply_yaml_flags(self.hd, external_flags, unflag_first=overwrite_flags)
+        else:
+            raise ValueError(f"{type} is an invalid type! Must be 'yaml' or 'uvflag'.")
+        # re-flag fully flagged baselines if necessary
+        if overwrite_flags:
+            for bl in full_flags:
+                tinds = self.hd.antpair2ind(bl)
```

Comment:
codecov says this is uncovered btw

Reply:
Thanks for pointing this out. I've added some coverage.